### DISCUSSION TITLE
关于认证系统集成问题

### DISCUSSION THREAD

**Discussion by JackeyTang** [PKUHPC/SCOW#11] [General]
作者 你好，我按照文档（https://pkuhpc.github.io/SCOW/ ）进行部署，遇到了一些问题。一开始我下载了github上的源码，安装好必要的docker环境，成功构建了镜像。然后我开始尝试着运行这些镜像，但是当我集成auth镜像的时候发现容器启动报错，根据报错信息，应该是要配置ldap认证的方式，所以我又找到了文档中测试用的ldap镜像，于是我构建并运行了该镜像，再次进行了测试发现，auth同样是启动报错，同样是没有配置认证方式，此时我想到是否是要配置相应的地址。让auth知道ldap服务在哪，于是我便找到了apps/auth/env/.env.test并修改了对应的地址（我把ldap集成到了docker-compose.yml，使用的地址便是服务的名称），之后我再次尝试启动auth发现仍然无法解决该问题

**Comment by ddadaal** [comment]
容器的环境变量都是配置在docker-compose.yml文件中对应部分的environment部分。请记住容器构建好之后不需要修改任何源码目录中的文件。

https://pkuhpc.github.io/SCOW/docs/common/auth/ldap#%E9%83%A8%E7%BD%B2%E5%B9%B6%E9%85%8D%E7%BD%AEldap%E8%AE%A4%E8%AF%81%E6%9C%8D%E5%8A%A1

**Reply by JackeyTang** [reply]
谢谢  我解决了这个问题  同时访问到了登陆页面，但是我却无法通过(用户：Manager 密码: admin)登陆。这是我的配置项：
![image](https://user-images.githubusercontent.com/45754513/170172242-bba350f2-4d49-40bf-9bc2-99dceb3dccaf.png)
请问是我的配置有什么问题吗？

**Reply by ddadaal** [reply]
目前无法以manager登录是正常的。请仔细阅读LDAP文档中的登录部分，搜索的用户是`{LDAP_ATTR_UID}=用户名`的节点，默认配置中`LDAP_ATTR_UID`为`uid`，所以如果以Manager登录，服务器会尝试搜索`LDAP_SEARCH_BASE`下`uid=Manager`的节点。但是LDAP镜像中的创建的Manager用户没有uid属性，只有cn属性=Manager，所以无法登录。

Manager用户本意是用来bind服务器和创建节点的，不应该被直接登录。

如果您觉得有需要，我可以再给LDAP镜像中多创建一个用来测试登录的用户。
